File: 3.py
# -*- coding:utf-8 -*-
# 题目: 从尾到头打印链表


class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    # 返回从尾部到头部的列表值序列，例如[1, 2, 3]
    def printListFromTailToHead(self, listNode):
        # write code here
        if listNode is None:
            return []
        result = []
        while listNode is not None:
            result.append(listNode.val)
            listNode = listNode.next
        return result[::-1]


oListNode = ListNode(3)
oListNode.next = ListNode(2)
oListNode.next.next = ListNode(1)
obj = Solution()
print(obj.printListFromTailToHead(oListNode))

File: 0-9/3.go
package main

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

/**
 * 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
 *
 * @param head ListNode类
 * @return int整型一维数组
 */
func printListFromTailToHead(head *ListNode) []int {
	// write code here
	result := []int{}
	for head != nil {
		result = append([]int{head.Val}, result...)
		head = head.Next
	}

	return result
}

func main() {
	head := &ListNode{
		Val: 67,
		Next: &ListNode{
			Val: 0,
			Next: &ListNode{
				Val: 24,
				Next: &ListNode{
					Val:  58,
					Next: nil,
				},
			},
		},
	}
	fmt.Println(printListFromTailToHead(head))
}
